src/utils/collection-utils.ts

```diff
@@ -52,3 +52,7 @@ export function transpose<T extends Record<string, string>>(obj: T): Transposed<
     }
     return result;
 }
+
+export function mapEntries<T extends Record<string, unknown>, R>(obj: T, mapper: (value: T[keyof T]) => R): Record<keyof T, R> {
```

Comment:
Maybe rename this function to `mapValues`, since it is similar to the `lodash` function.